network/secondary_endpoint_client_linux.go

```diff
@@ -0,0 +1,153 @@
+		netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+		ep:             endpoint,
+	}
+
+	return client
+}
+
+func (client *SecondaryEndpointClient) AddEndpoints(epInfo *EndpointInfo) error {
+	iface, err := client.netioshim.GetNetworkInterfaceByMac(epInfo.MacAddress)
+	if err != nil {
+		return newErrorSecondaryEndpointClient(err)
+	}
+
+	epInfo.IfName = iface.Name
+	if _, exists := client.ep.SecondaryInterfaces[iface.Name]; exists {
+		return newErrorSecondaryEndpointClient(errors.New(iface.Name + " already exists"))
+	}
+	client.ep.SecondaryInterfaces[iface.Name] = &InterfaceInfo{
+		Name:              iface.Name,
+		MacAddress:        epInfo.MacAddress,
+		IPAddress:         epInfo.IPAddresses,
+		NICType:           epInfo.NICType,
+		SkipDefaultRoutes: epInfo.SkipDefaultRoutes,
+	}
+
+	return nil
+}
+
+func (client *SecondaryEndpointClient) AddEndpointRules(_ *EndpointInfo) error {
+	return nil
+}
+
+func (client *SecondaryEndpointClient) DeleteEndpointRules(_ *endpoint) {
+}
+
+func (client *SecondaryEndpointClient) MoveEndpointsToContainerNS(epInfo *EndpointInfo, nsID uintptr) error {
+	// Move the container interface to container's network namespace.
+	logger.Info("[net] Setting link %v netns %v.", zap.String("IfName", epInfo.IfName), zap.String("NetNsPath", epInfo.NetNsPath))
+	if err := client.netlink.SetLinkNetNs(epInfo.IfName, nsID); err != nil {
+		return newErrorSecondaryEndpointClient(err)
+	}
+
+	return nil
+}
+
+func (client *SecondaryEndpointClient) SetupContainerInterfaces(epInfo *EndpointInfo) error {
+	logger.Info("[net] Setting link state up.", zap.String("IfName", epInfo.IfName))
+	if err := client.netlink.SetLinkState(epInfo.IfName, true); err != nil {
+		return newErrorSecondaryEndpointClient(err)
+	}
+
+	return nil
+}
+
+func (client *SecondaryEndpointClient) ConfigureContainerInterfacesAndRoutes(epInfo *EndpointInfo) error {
+	if err := client.netUtilsClient.AssignIPToInterface(epInfo.IfName, epInfo.IPAddresses); err != nil {
+		return newErrorSecondaryEndpointClient(err)
+	}
+
+	ifInfo, exists := client.ep.SecondaryInterfaces[epInfo.IfName]
+	if !exists {
+		return newErrorSecondaryEndpointClient(errors.New(epInfo.IfName + " does not exist"))
+	}
+
+	if len(epInfo.Routes) < 1 {
+		return newErrorSecondaryEndpointClient(errors.New("routes expected for " + epInfo.IfName))
+	}
+
+	if err := addRoutes(client.netlink, client.netioshim, epInfo.IfName, epInfo.Routes); err != nil {
+		return newErrorSecondaryEndpointClient(err)
+	}
+
+	ifInfo.Routes = append(ifInfo.Routes, epInfo.Routes...)
+
+	return nil
+}
+
+func (client *SecondaryEndpointClient) DeleteEndpoints(ep *endpoint) error {
+	if ep.NetworkNameSpace != "" {
+		// Get VM namespace
+		vmns, err := netns.New().Get()
+		if err != nil {
+			return newErrorSecondaryEndpointClient(err)
+		}
+
+		// Open the network namespace.
+		logger.Info("Opening netns", zap.Any("NetNsPath", ep.NetworkNameSpace))
+		ns, err := OpenNamespace(ep.NetworkNameSpace)
```

Comment:
if network namespace doesn't exist, it should not throw error. can you check for not exist error and ignore

Reply:
will check for not exist error in delete and return nil